src/esp/physics/RigidBase.h

```diff
@@ -598,6 +610,10 @@ class RigidBase : public Magnum::SceneGraph::AbstractFeature3D {
    */
   scene::SceneNode* visualNode_ = nullptr;
 
+  //! all nodes created when this object's render asset was added to the
+  //! SceneGraph
+  std::vector<esp::scene::SceneNode*> visualNodes_;
```

Comment:
Wait, is this whole thing actually needed? Contents of this vector is the same as if you would recursively collect all children of `visualNode_`, no?

I'm just trying to minimize the amount of redundant state that needs to be maintained.

Reply:
True for now. 👍  However, we are planning SceneGraph re-structuring in the near future which will invalidate this assumption, so planning for that.